diesel/src/pg/expression/functions.rs

```diff
@@ -748,3 +748,34 @@ define_sql_function! {
     /// ```
     fn array_append<Arr: ArrayOrNullableArray<Inner=T> + SingleValue, T: SingleValue>(a: Arr, e: T) -> Array<T>;
 }
+
+#[cfg(feature = "postgres_backend")]
+define_sql_function! {
+    /// Returns a text representation of the array's dimensions
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # include!("../../doctest_setup.rs");
+    /// #
+    /// # fn main(){
+    /// #    run_test().unwrap();
+    /// # }
+    /// # fn run_test()->QueryResult<()>{
+    /// #   use diesel::dsl::array_dims;
+    /// #   use diesel::sql_types::{Nullable,Array,Integer};
+    /// #   let connection = &mut establish_connection();
+    ///
+    /// let dims = diesel::select(array_dims::<Array<_>,_,_>(vec![1,2]))
+    ///     .get_result::<String>(connection).unwrap();
+    /// assert!(String::from("[1:2]").eq(&dims));
+    ///
+    /// let dims = diesel::select(array_dims::<Array<_>,_,_>(vec![vec![1,2,3],vec![4,5,6]]))
+    /// .get_result::<String>(connection).unwrap();
+    /// assert!(String::from("[1:2][1:3]").eq(&dims));
```

Comment:
The crazy error here say that diesel didn't know how to transform `Vec<Vec<_>>` to 'Array<_>` becase Diesel does not support multidimensional arrays yet. So you can remove this test for now.